packages/flutter/lib/src/services/keyboard_key.dart

```diff
@@ -0,0 +1,2543 @@
+
+  /// A unique code representing this key.
+  ///
+  /// This is an opaque code, and should not be unpacked to derive information
+  /// from it, as the representation of the code could change at any time.
+  final int keyId;
+
+  /// The debug string to print for this keyboard key, which will be null in
+  /// release mode.
+  final String debugName;
+
+  /// The Unicode string representing the character produced by a [RawKeyEvent].
+  ///
+  /// This value is useful for describing or matching mnemonic keyboard
+  /// shortcuts.
+  ///
+  /// On most platforms this is a single code point, but it could contain any
+  /// Unicode string. The `keyLabel` differs from [RawKeyEvent.character]
+  /// because `keyLabel` only takes into account the key being pressed, not any
+  /// combining keys pressed before it, so, for example, an “o” that follows a
+  /// combining dieresis (“¨”, COMBINING DIAERESIS (U+0308)) would just return
+  /// “o” for [keyLabel], but would return “ö” for [RawKeyEvent.character].
+  ///
+  /// {@macro flutter.services.RawKeyEventData.keyLabel}
+  final String keyLabel;
+
+  @override
+  int get hashCode => keyId.hashCode;
+
+  @override
+  bool operator ==(dynamic other) {
+    if (other.runtimeType != runtimeType) {
+      return false;
+    }
+    final LogicalKeyboardKey typedOther = other;
+    return keyId == typedOther.keyId;
+  }
+
+  /// Finds the [LogicalKeyboardKey] that matches the given ID.
+  static LogicalKeyboardKey findKeyByKeyId(int keyId) => _knownLogicalKeys[keyId];
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(StringProperty('keyId', '0x${keyId.toRadixString(16).padLeft(8, '0')}', showName: true));
+    properties.add(StringProperty('keyLabel', keyLabel, showName: true));
+    properties.add(StringProperty('debugName', debugName, showName: true, defaultValue: null));
+  }
+
+  /// Returns true if the given label represents a Unicode control character.
+  ///
+  /// Examples of control characters are characters like "U+000A LINE FEED (LF)"
+  /// or "U+001B ESCAPE (ESC)".
+  ///
+  /// See <https://en.wikipedia.org/wiki/Unicode_control_characters> for more
+  /// information.
+  ///
+  /// Used by [RawKeyEvent] subclasses to help construct IDs.
+  static bool isControlCharacter(String label) {
+    if (label.length > 1) {
+      return false;
+    }
+    final int codeUnit = label.codeUnitAt(0);
+    return (codeUnit <= 0x1f && codeUnit >= 0x00) || (codeUnit >= 0x7f && codeUnit <= 0x9f);
+  }
+
+  /// Returns true if the [keyId] of this object is one that is autogenerated by
+  /// Flutter.
+  ///
+  /// Autogenerated key IDs are generated in response to platform key codes
+  /// which Flutter doesn't recognize, and their IDs shouldn't be used in a
+  /// persistent way.
+  ///
+  /// Autogenerated IDs should be a rare occurrence: Flutter supports most keys.
```

Comment:
If somebody finds an unsupported key - is that a bug? should they file a bug report?

Reply:
Hmm. I guess?  I'll add a note to file a bug in that case...